### PR TITLE
Tapir based secuirty

### PR DESCRIPTION
**Pull request by Pask423**

### CODE REVIEW

**Comment by Pask423** [comment]
closes #769

**Comment by Pask423** [comment]
closes #793

File: backend/src/main/scala/com/softwaremill/bootzooka/user/UserApi.scala [hunk 1]
```diff
@@ -44,36 +44,36 @@ class UserApi(http: Http, auth: Auth[ApiKey], userService: UserService, xa: Tran
     }
 
   private val changePasswordEndpoint = secureEndpoint.post
-    .in(UserPath / "changepassword")
+    .securityIn(UserPath / "changepassword")
     .in(jsonBody[ChangePassword_IN])
     .out(jsonBody[ChangePassword_OUT])
-    .serverLogic { case (authData, data) =>
+    .serverSecurityLogic(authData => auth.applyEither(authData).map(_.left.map(exceptionToErrorOut)))
```
Comment (adamw) [inline backend/src/main/scala/com/softwaremill/bootzooka/user/UserApi.scala:50]: can't we use `.toOut` here as for server logic?

Comment (Pask423) [reply backend/src/main/scala/com/softwaremill/bootzooka/user/UserApi.scala:50]: Not if I understand it correctly here we just mapping left while `.toOut` perform mapping on both left and right

Comment (Pask423) [reply backend/src/main/scala/com/softwaremill/bootzooka/user/UserApi.scala]: I didn't understand correctly fixed

File: backend/src/main/scala/com/softwaremill/bootzooka/user/UserApi.scala [hunk 2]
```diff
@@ -44,36 +44,39 @@ class UserApi(http: Http, auth: Auth[ApiKey], userService: UserService, xa: Tran
...
+    .serverSecurityLogic(authData => auth(authData).toOut)
+    .serverLogic(id =>
+      data =>
+        (for {
+          _ <- userService.changePassword(id, data.currentPassword, data.newPassword).transact(xa)
+        } yield ChangePassword_OUT()).toOut
+    )
 
   private val getUserEndpoint = secureEndpoint.get
     .in(UserPath)
```
Comment (adamw) [inline backend/src/main/scala/com/softwaremill/bootzooka/user/UserApi.scala:59]: if `changePasswordEndpoint` uses `.securityIn` for the path, shouldn't we use this here as well?

Comment (Pask423) [reply backend/src/main/scala/com/softwaremill/bootzooka/user/UserApi.scala:59]: Now I am not sure if `changePasswordEndpoint` require path in `.securityIn` , which way is better in your opinion ?

Comment (adamw) [reply backend/src/main/scala/com/softwaremill/bootzooka/user/UserApi.scala:59]: yeah you're right it's fine to keep them all in the regular inputs. The path matching is done on both anyway

**Comment by Pask423** [comment]
closes #796
Probably also can close #270 because newest version of otj-pg-embedded includes tests containers.
@adamw could you verify if such setting is enough to close #270 ?

**Comment by adamw** [comment]
Thanks :)

**Comment by adamw** [comment]
@Pask423 so right now postgres is run through a container?

**Comment by adamw** [comment]
@Pask423 indeed, it seems #270 is done also :)

**Pull request merged**